src/date/format-properties.js

```diff
@@ -208,6 +209,58 @@ return function( pattern, cldr ) {
 
 			// Zone
 			case "z":
+
+				//z...zzz: "{shortRegion}", eg. "PST" or "PDT".
+				//zzzz: "{regionName} {Standard Time}" or "{regionName} {Daylight Time}",
+				//eg. "Pacific Standard Time" or "Pacific Daylight Time".
+				if ( properties.timeZone ) {
+
+					// The latest metazone data of the metazone array.
+					//TODO expand to support the historic metazones based on the given date.
+					var standardTzName,
+						daylightTzName,
+						metaZone = cldr.supplemental([
+							"metaZones/metazoneInfo/timezone", properties.timeZone, 0,
+							"usesMetazone/_mzone"
+						]);
+
+					if ( metaZone ) {
+						standardTzName = cldr.main([
+							"dates/timeZoneNames/metazone",
+							metaZone,
+							length < 4 ? "short" : "long",
+							"standard"
+						]);
+						daylightTzName = cldr.main([
+							"dates/timeZoneNames/metazone",
+							metaZone,
+							length < 4 ? "short" : "long",
+							"daylight"
+						]);
+						if ( !standardTzName && !daylightTzName ) {
+							var exemplarCity = cldr.main(
+								"dates/timeZoneNames/zone/",
+								properties.timeZone,
+								"/exemplarCity"
+							);
+							standardTzName = cldr.main(
+								"dates/timeZoneNames/regionFormat-type-standard"
+							)
+							.replace( /\{0\}/, exemplarCity );
```

Comment:
Instead of replacing `{0}` yourself, you can use [src/common/format-message](https://github.com/globalizejs/globalize/blob/master/src/common/format-message.js) ([example](https://github.com/globalizejs/globalize/blob/master/src/date/expand-pattern.js#L32-L37)).